rfc-120-csv-preview.md

```diff
@@ -0,0 +1,134 @@
+# CSV Preview
+
+[govuk-content-schemas]: https://github.com/alphagov/govuk-content-schemas
+[government-prefix]: https://trello.com/c/74KTVXBs/52-whitehall-relies-on-prefix-government-and-world-routes
+[router]: https://github.com/alphagov/router#backend-handler
+
+## Summary
+
+Move the CSV Preview feature out of Whitehall and into Government Frontend, so that it can be used for documents published by other apps, like Content Publisher. The new implementation will work by rendering a content item in a new `attachment_preview` format, which represents an attachment to preview.
+
+## Problem
+
+A user viewing a Whitehall document on GOV.UK can click a "View Online" link for any of its CSV file attachments. This takes the user to a new page, currently rendered by Whitehall. The page contains:
+
+   - A 'header' section with details of the parent document (and navigation links)
+   - A 'body' section with a preview of the content of the CSV attachment
+
+This CSV Preview feature only works for CSV file attachments uploaded through Whitehall. We need it to work for CSV attachments published through Content Publisher, which is superseding Whitehall.
+
+## Proposal
+
+### Step 1
+
+Create a new `attachment_preview` format in [govuk-content-schemas][] to represent the preview page for an attachment. A publishing app will publish content items in this format for each CSV attachment to preview.
+
+```
+# formats/attachment_preview.jsonnet
+(import "shared/default_format.jsonnet") + {
+  document_type: "attachment_preview",
+  attachment: {
+    "$ref": "#/definitions/file_attachment_asset"
+  },
+  edition_links: {
+    parent: {
+      description: "The parent content item.",
+      maxItems: 1,
+    },
+  }
+}
+```
+
+- Using a content item is consistent with the one-to-one mapping of pages and content on GOV.UK. This should make it easier to debug issues with the feature, since it follows an existing pattern.
+
+- The schema includes a link to the parent document. Using an expanded link means we don't need to make an extra API call to the Content Store in order to show information about the parent document.
```

Comment:
We will also include a link to the primary organisation? I can't remember if some things fully expect that or not. I suppose we have precedent in html publications to follow.

Reply:
I don't _think_ we _need_ it. However, I found a couple of reasons to include it (updated).

> Including organisation info is [a longterm endeavour](https://github.com/alphagov/govuk-rfcs/pull/92), and will also make it a bit simpler to render the organisation logo in the header.